### ISSUE TITLE
tinytex is unable to display "contents"

### ISSUE THREAD

**Issue by Sakura-gh** [yihui/tinytex#220]
I want to use tinytex to write a report with contents, as we all kown, in latex we should use `\tableofcontents` to produce contents. But unfortunately, my tinytex cannot do that. I've used the same code in overleaf and produced contents successfully, maybe something wrong happened. Please help me!
my code:
~~~latex
%! TEX program=xelatex
\documentclass[UTF8,11pt]{article} % 文档类型
\usepackage{ctex} % 中文
\usepackage[T1]{fontenc} % 加载宏包 fontenc 防止某些符号显示不正常
\usepackage[margin=1.2in]{geometry} % 设置纸张大小和纸边距
\usepackage{amsmath} % 加载数学宏包
\usepackage{amssymb}
\usepackage{graphicx} % 用于添加图片
% \graphicspath{{img/}} % 指定图片所在文件夹
\usepackage{float} % 图片浮动
\usepackage{titletoc} % 目录格式
\usepackage{hyperref} % 目录跳转
\hypersetup{ % 设置目录跳转及对应的颜色 
colorlinks=true,
linkcolor=black
}
\renewcommand\contentsname{目录} % 将英文名contents改为中文名"目录"
\setlength{\parindent}{0pt} % 段首不缩进

\begin{document}

\newpage
\tableofcontents
\thispagestyle{empty}

\newpage
\setcounter{page}{1}
\section{实验目的}
设计并实现一个精简型单用户SQL引擎(DBMS)MiniSQL，允许用户通过字符界面输入SQL语句实现：
\begin{itemize}
    \item[1)] 表的建立/删除
    \item[2)] 索引的建立/删除
    \item[3)] 表记录的插入/删除/查找
\end{itemize}
通过对MiniSQL的设计与实现，提高学生的系统编程能力，加深对数据库系统原理的理解。\\

\section{系统需求}
本节将简要介绍minisql系统的一些基本需求，其中着重突出本人主要参与的Index Manager模块说明。
\subsection{需求概述}
\subsubsection{数据类型}
只要求支持三种基本数据类型：int，char(n)，float，其中char(n)满足1 <= n <= 255。
\subsubsection{表定义}
一个表最多可以定义32个属性，各属性可以指定是否为unique；支持unique属性的主键定义。
\subsubsection{索引的建立和删除}
对于表的主键自动建立B+树索引，对于声明为unique的属性可以通过SQL语句由用户指定建立/删除B+树索引（因此，所有的B+树索引都是单属性单值的）。
\subsubsection{查找记录}
可以通过指定用and连接的多个条件进行查询，支持等值查询和区间查询。
\subsubsection{插入和删除记录}
1、支持每次一条记录的插入操作；\\
2、支持每次一条或多条记录的删除操作。(where条件是范围时删除多条)\\

\section{实验环境}

\section{模块设计}

\section{模块实现}

\section{遇到的问题及解决方法}

\section{总结}

\end{document}

~~~
the result pdf produced by tinytex:
![image](https://user-images.githubusercontent.com/50900131/84843281-0b482e80-b07a-11ea-992c-89c73dd2d588.png)
the result pdf produce by overleaf:
![image](https://user-images.githubusercontent.com/50900131/84843311-231fb280-b07a-11ea-9bba-04af39ef24f2.png)
 
---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('tinytex')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('yihui/tinytex')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by yihui** [comment]
Please follow the issue guide to provide a full reproducible example first. With a fragment of LaTeX code, I can't compile it, so there is no way for me to help you.

**Comment by Sakura-gh** [comment]
> Please follow the issue guide to provide a full reproducible example first. With a fragment of LaTeX code, I can't compile it, so there is no way for me to help you.

sorry for that, I've updated the issue with full code.

**Comment by yihui** [comment]
I compiled it via

```r
tinytex::xelatex('test.tex')
```

and it worked fine for me: [test.pdf](https://github.com/yihui/tinytex/files/4790215/test.pdf) The table of contents was correctly generated. How did you compile your document?

**Comment by Sakura-gh** [comment]
Fine. I think I've got why it doesn't work. In VScode, I used the extension 'LaTex Workshop', it offers the command 'xelatex', but when I clicked,  it'll just complie once.
But when I complie in terminal
~~~latex
xelatex main.tex
~~~
it gives me warning: `No file main.aux.`. I realised that I need to use `xelatex main.tex` twice so that the contents will generated successfully.
Anyway, thanks for yihui's help! Also thanks for **tinytex**, it helps me to understand latex more deeply!
